File: cs50/pset6/greedy.py
import cs50

def GetPositiveFloatWithMessage(message):
    
    while True:

        print(message)
        
        number = cs50.get_float()
        
        if number >= 0:
            break
        
        print("Please type the positive number or zero!");
        
    return number


print("O hai! ")
    
cents = 100 * GetPositiveFloatWithMessage("How much change is owed?")
    
coins = 0

for i in [25, 10, 5, 1]:
    if cents >= i:
        coins += cents // i
        cents = cents % i
    

print(coins)

File: cs50/pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float GetPositiveFloatWithMessage(string);
int DollarsToCents(float);

int main(void){
    
    printf("O hai! ");
    
    int cents = DollarsToCents( GetPositiveFloatWithMessage("How much change is owed?\n") );
    
    int coins = 0;

    while(cents >= 25){
        cents-=25; coins++;
    }

    while(cents >= 10){
        cents-=10; coins++;
    }

    while(cents >= 5){
        cents-=5; coins++;
    }

    while(cents >= 1){
        cents--; coins++;
    }

    printf("%i\n", coins);
    
    return 0;
}



float GetPositiveFloatWithMessage(string message){
    
    float number;
    bool flag;
    
    do{

        printf("%s", message);
        number = GetFloat();
        
        flag = (number >= 0);
        
        if(!flag){
            printf("Please type the positive number or zero!\n");
        }
        
    } while(!flag);
    
    return number;
    
}

int DollarsToCents(float dollars){
    return (int) roundf(dollars * 100);
}